src/hyperlink/_url.py

```diff
@@ -701,22 +747,24 @@ def parse_host(host):
     """
     if not host:
         return None, u''
+
     if u':' in host:
         try:
-            inet_pton(socket.AF_INET6, host)
+            inet_pton(AF_INET6, host)
         except socket.error as se:
             raise URLParseError('invalid IPv6 host: %r (%r)' % (host, se))
         except UnicodeEncodeError:
             pass  # TODO: this can't be a real host right?
         else:
-            family = socket.AF_INET6
-            return family, host
-    try:
-        inet_pton(socket.AF_INET, host)
-    except (socket.error, UnicodeEncodeError):
-        family = None  # not an IP
+            family = AF_INET6  # type: Optional[AddressFamily]
     else:
-        family = socket.AF_INET
+        try:
+            inet_pton(AF_INET, host)
+        except (socket.error, UnicodeEncodeError):
+            family = None  # not an IP
+        else:
+            family = AF_INET
+
```

Comment:
Some re-arranging here so that the only `return` statement is below, and this code just sets those values.

Reply:
This used to be my style for a long time (mostly due to a harsh-grading TA), then I found guard statements fit my brain better (and bought me back indentation levels). Would love to discuss the single-return practice sometime.

Reply:
I don't object to either, but the code was doing both, so I figured it would be simpler to pick one or the other for consistency.  In this case I thought "figure out what family is" would be one step and "return a result with family and host" would follow.  Alternatively, we could have changed the `family = FOO` lines to `return FOO, host`.